components/genome-browser-facets.js

```diff
@@ -0,0 +1,536 @@
+                  />
+                );
+              })}
+              <button
+                className="flex"
+                onClick={() => this.clearGenomeFilters()}
+              >
+                <XCircleIcon className="h5 w-5" />
+                Clear all filters
+              </button>
+            </div>
+            {this.props.filteredFiles.length === 0 ? (
+              <div>
+                No files match the selected filters. Try different filters to
+                visualize results.
+              </div>
+            ) : null}
+          </React.Fragment>
+        ) : (
+          <div className="mb-4">
+            To select a filter, make a selection from &quot;Refine your
+            search&quot; below.
+          </div>
+        )}
+        <Button className="flex mb-1" onClick={this.toggleFacetDisplay}>
+          {this.state.facetDisplay ? (
+            <ChevronDownIcon className="h-5 w-5" />
+          ) : (
+            <ChevronRightIcon className="h-5 w-5" />
+          )}
+          <span>Refine your search</span>
+        </Button>
+        {this.state.facetDisplay ? (
+          <div className="grid grid-cols-5 border bg-gray-100 rounded-md gap-2 mb-4">
+            {facetParameters.map((facet, facetIndex) => {
+              if (facet.typeahead) {
+                return (
+                  <Facet
+                    typeahead
+                    key={facetIndex}
+                    facetTitle={facet.title}
+                    facetName={facet.type}
+                    facetArray={facetObject[facet.type]}
+                    handleSearch={this.handleSearch}
+                    addGenomeFilter={this.addGenomeFilter}
+                    selectedFacets={this.state.selectedFacets}
+                    unsanitizedSearchTerm={
+                      this.state.unsanitizedSearchTerms[facet.type]
+                    }
+                  />
+                );
+              }
+              return (
+                <Facet
+                  typeahead={false}
+                  key={facetIndex}
+                  facetTitle={facet.title}
+                  facetName={facet.type}
+                  facetArray={facetObject[facet.type]}
+                  addGenomeFilter={this.addGenomeFilter}
+                  selectedFacets={this.state.selectedFacets}
+                />
+              );
+            })}
+          </div>
+        ) : null}
+      </React.Fragment>
+    );
+  }
+}
+
+GenomeBrowserFacets.propTypes = {
+  files: PropTypes.array.isRequired,
+  handleFacetList: PropTypes.func.isRequired,
+  filteredFiles: PropTypes.array.isRequired,
+  selectedFilters: PropTypes.array.isRequired,
+};
+
+// Generate facets for the genome browser view
+export function createFacets(
+  files,
+  filteredFiles,
+  searchTerms,
+  selectedFilters
+) {
+  // initialize facet object
+  const facetObject = {};
+  facetParameters.forEach((facet) => {
+    facetObject[facet.type] = [];
+  });
```

Comment:
Since `facetParameters` is statically initialized, it seems like `facetObject` could also be statically initialized. The way it is, facetObject gets the same value every time `createFacets` is called no matter what values get passed in its parameters.